### ISSUE TITLE
Update build to use dotnet global tools.

### ISSUE THREAD

**Issue by michael-emmi** [smackers/smack#531]
Boogie and Corral releases are now being pushed to nuget.org as [.NET Core Global Tools](https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools):
* [Boogie](https://www.nuget.org/packages/Boogie/)
* [Corral](https://www.nuget.org/packages/Corral/)

These artifacts can be installed rather than rebuilt like so:
```
$ dotnet tool install --global Boogie
```
although we’d probably prefer to install specific versions like so:
```
$ dotnet tool install --global Boogie --version 2.4.4
```
Doing this will require having .NET Core installed, replacing our dependency on Mono.

**Comment by zvonimir** [comment]
To completely get rid of the dependency on mono, we would need to do something similar with lockpwn and symbooglix too. I would guess that should not be that hard.
Btw, thanks for doing this! It'll be good not to have to build all of these from scratch every time SMACK is being installed.

**Comment by michael-emmi** [comment]
> To completely get rid of the dependency on mono, we would need to do something similar with lockpwn and symbooglix too. I would guess that should not be that hard.

@zvonimir yes indeed — perhaps some students would like to be involved in this?

**Comment by zvonimir** [comment]
I addressed this in #563
